src/Type/Php/ArraySumFunctionDynamicReturnTypeExtension.php

```diff
@@ -36,12 +36,20 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		}
 
 		if ($arrayType->isIterableAtLeastOnce()->yes()) {
-			return TypeCombinator::intersect($itemType, new UnionType([new IntegerType(), new FloatType()]));
+			if ($itemType instanceof IntegerType || $itemType instanceof FloatType) {
+				return $itemType;
+			}
+
+			return new UnionType([new IntegerType(), new FloatType()]);
+		}
+
+		if ($itemType instanceof IntegerType || $itemType instanceof FloatType) {
+			return TypeCombinator::union(new ConstantIntegerType(0), $itemType);
 		}
 
 		return TypeCombinator::union(
 			new ConstantIntegerType(0),
-			TypeCombinator::intersect($itemType, new UnionType([new IntegerType(), new FloatType()]))
+			new UnionType([new IntegerType(), new FloatType()])
```

Comment:
I don't think it makes sense to modify this.

Reply:
For an array of string, $itemType is a stringType.
So `TypeCombinator::intersect($itemType, new UnionType([new IntegerType(), new FloatType()])` is nothing.
That's why the dynamic return type return  `new ConstantIntegerType(0)`.

The intersect seems to be the reason of the bug.

Reply:
But on the other hand, this intersect allows us to specify that when we only have `array<int>`, the result is going to be `int`.

Reply:
Isn't this handle by the previous check
```
if ($intUnionFloat->isSuperTypeOf($itemType)->yes()) {
    return TypeCombinator::union(new ConstantIntegerType(0), $itemType);
}
```
?